backend_test.go

```diff
@@ -0,0 +1,290 @@
+	configData := map[string]interface{}{
+		"configName":  HOME_TENANCY_ID_CONFIG_NAME,
+		"configValue": homeTenancyId,
+	}
+
+	configReq := &logical.Request{
+		Operation: logical.CreateOperation,
+		Storage:   backendConfig.StorageView,
+		Data:      configData,
+	}
+
+	configReq.Path = configPath
+	resp, err = backend.HandleRequest(context.Background(), configReq)
+	if err != nil || (resp != nil && resp.IsError()) {
+		t.Fatalf("bad: config creation failed. resp:%#v\n err:%v", resp, err)
+	}
+}
+
+func initTest(t *testing.T) (backend logical.Backend, config *logical.BackendConfig, err error) {
+
+	defaultLeaseTTLVal := time.Hour * 24
+	maxLeaseTTLVal := time.Hour * 24 * 2
+
+	config = &logical.BackendConfig{
+		Logger: nil,
+		System: &logical.StaticSystemView{
+			DefaultLeaseTTLVal: defaultLeaseTTLVal,
+			MaxLeaseTTLVal:     maxLeaseTTLVal,
+		},
+		StorageView: &logical.InmemStorage{},
+	}
+
+	backend, err = Factory(context.Background(), config)
+
+	if err != nil {
+		return
+	}
+
+	createHomeTenancy(t, config, backend)
+
+	roleOcidList := os.Getenv("ROLE_OCID_LIST")
+	if roleOcidList == "" {
+		return nil, nil, fmt.Errorf("ROLE_OCID_LIST environment variable is empty")
+	}
+	//Create the devRole
+	devRoleData := map[string]interface{}{
+		"role":            DEV_ROLE,
+		"description":     DEV_ROLE + " description",
+		"add_ocid_list":   roleOcidList,
+		"add_policy_list": "policy1,policy2",
+		"ttl":             1500,
+	}
+
+	err = createRole(devRoleData, DEV_ROLE, backend, config)
+	if err != nil {
+		return
+	}
+
+	//Create the opsRole
+	opsRoleData := map[string]interface{}{
+		"role":            OPS_ROLE,
+		"description":     OPS_ROLE + " description",
+		"add_ocid_list":   "",
+		"add_policy_list": "policy3",
+		"ttl":             1000,
+	}
+
+	err = createRole(opsRoleData, OPS_ROLE, backend, config)
+	if err != nil {
+		return
+	}
+
+	//Create the knowledgeWorkerRole
+	knowledgeWorkerRole := map[string]interface{}{
+		"role":            KNOWLEDGE_WORKER_ROLE,
+		"description":     KNOWLEDGE_WORKER_ROLE + " description",
+		"add_ocid_list":   roleOcidList,
+		"add_policy_list": "policy1,policy5",
+		"ttl":             1000,
+	}
+
+	err = createRole(knowledgeWorkerRole, KNOWLEDGE_WORKER_ROLE, backend, config)
+	if err != nil {
+		return
+	}
+
+	return
+}
+
+func createRole(roleData map[string]interface{}, roleName string, backend logical.Backend, config *logical.BackendConfig) error {
+	roleRequest := &logical.Request{
+		Operation: logical.CreateOperation,
+		Storage:   config.StorageView,
+		Data:      roleData,
+	}
+
+	roleRequest.Path = "role/" + roleName
+	response, err := backend.HandleRequest(context.Background(), roleRequest)
+	if err != nil || (response != nil && response.IsError()) {
+		return fmt.Errorf("bad: role creation failed. resp:%#v\n err:%v", response, err)
+	}
+
+	return nil
+}
+
+func TestBackEnd_ValidateUserApiKeyLogin(t *testing.T) {
+
+	// Skip tests if we are not running acceptance tests
```

Comment:
Can we add some explanation around this?

Reply:
This is a common pattern in Vault tests. Acceptance tests usually require env variables or other preconfiguration for the test to succeed, so you only get those if you opt in with `VAULT_ACC=1 go test ...`.  The `VAULT_ACC` env var is a defacto standard for enabling them.